pkg/v3/resource/chartconfig/update.go

```diff
@@ -86,7 +86,11 @@ func (r *Resource) newUpdateChange(ctx context.Context, obj, currentState, desir
 		}
 
 		if isChartConfigModified(desiredChartConfig, currentChartConfig) {
-			chartConfigsToUpdate = append(chartConfigsToUpdate, desiredChartConfig)
+			// Make a copy and set the resource version so the CR can be updated.
+			chartConfigToUpdate := desiredChartConfig.DeepCopy()
```

Comment:
This works with the deep copy and is safer to avoid modifying the desired state.